### ISSUE TITLE
shlex.split isn't 100% appropriate for shebangs

### ISSUE THREAD

**Issue by asottile** [chriskuehl/identify#26]
It seems shebangs only allow double quoted strings, here's a shebang in the wild which would have difficulties being shlexed:

```
$ head -1 venv/bin/pip
#!/tmp/foo'bar/venv/bin/python
```

Trace when parsing:

```python
>>> identify.identify.parse_shebang(io.BytesIO(b"#!/tmp/foo'bar/venv/bin/python"))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "identify/identify.py", line 136, in parse_shebang
    cmd = tuple(shlex.split(first_line))
  File "/usr/lib/python2.7/shlex.py", line 279, in split
    return list(lex)
  File "/usr/lib/python2.7/shlex.py", line 269, in next
    token = self.get_token()
  File "/usr/lib/python2.7/shlex.py", line 96, in get_token
    raw = self.read_token()
  File "/usr/lib/python2.7/shlex.py", line 172, in read_token
    raise ValueError, "No closing quotation"
ValueError: No closing quotation
```

Originally from https://github.com/pre-commit/pre-commit/issues/625

**Comment by asottile** [comment]
Even quoted strings don't seem to work, though setuptools will write them.  I guess we should try `shlex.split` and fall back to `WS_RE.split`

**Comment by asottile** [comment]
This is now available in v1.0.6
